### PR TITLE
release build_runner_core 4.1.0

### PR DESCRIPTION
**Pull request by jakemac53**

### CODE REVIEW

**Comment by natebosch** [comment]
Can we wait for https://github.com/dart-lang/build/pull/2449 ?

**Comment by jakemac53** [comment]
Sure I am fine with that

File: build_runner_core/pubspec.yaml [hunk 1]
```diff
@@ -39,9 +39,5 @@ dev_dependencies:
     path: ../_test_common
 
 dependency_overrides:
-  build:
-    path: ../build
-  build_resolvers:
-    path: ../build_resolvers
   # https://github.com/dart-lang/sdk/issues/38499
   analyzer: 0.38.2
```
Comment (natebosch) [inline build_runner_core/pubspec.yaml]: I think you can't publish with a dependency_override like this? Are you planning on removing locally during the publish?

Comment (jakemac53) [reply build_runner_core/pubspec.yaml]: Yes I am removing locally before publish

**Pull request merged**